buildmaster/NMCPD/filter.py

```diff
@@ -0,0 +1,64 @@
+#Filter for NMCPD
+
+import yaml
+
+def filter_NMCPD():
+    with open('metadata.yaml', 'r') as file:
+        metadata = yaml.safe_load(file)
+
+    version = metadata['hepdata']['version']
+    tables  = metadata['hepdata']['tables']
+
+    data_central = []
+    kin_avg = []
+    kin_min = []
+    kin_max = []
+    stat_error = []
+    sys_error = []
+    index = 1
+    
+    for i in tables:
+        hepdata_tables="rawdata/HEPData-ins426595-"+version+"-Table_"+str(i)+".yaml"
+        with open(hepdata_tables, 'r') as file:
+            input = yaml.safe_load(file)
+            
+        values = input['dependent_variables'][0]['values']
+        x = float(input['dependent_variables'][0]['qualifiers'][4]['value'])
+        sqrts = float(input['dependent_variables'][0]['qualifiers'][3]['value'])
+        
+        for j in range(len(values)):
+            data_central_value = {'index' : index, 'value' : input['dependent_variables'][0]['values'][j]['value']}      
+            data_central.append(data_central_value)
+            Q2 = input['independent_variables'][0]['values'][j]['value']
+            y = Q2 / ( sqrts * sqrts * x )
+            kin_avg_value = {'index' : index , 'x' : x , 'q2' : Q2 , 'y' : y}
+            kin_avg.append(kin_avg_value)
+            kin_min_value = {'index' : index , 'x' : None , 'q2' : None , 'y' : None}
+            kin_min.append(kin_min_value)
+            kin_max_value = {'index' : index , 'x' : None , 'q2' : None , 'y' : None}
+            kin_max.append(kin_max_value)
```

Comment:
In the `None` case (if there is expected to be a  `None` case) I would rather have the minimum and the maximum to be equal to the central value (or the entire dictionary set to `None`).
Having the `yaml` file fill with `None`s seems a bit like a waste.

Reply:
Done.